### ISSUE TITLE
Feature request: normalize for tpf.interact()

### ISSUE THREAD

**Issue by MNGuenther** [KeplerGO/lightkurve#578]
I would like to request a "normalize" feature for tpf.interact(), to be passed as an argument when calling the function: currently the y axis is shown as "Flux (e/s)". It would be nice to be able to switch this view to a "normalized" lightcurve, e.g. simply divided by the median flux. This is particularly important when comparing the relative depth/amplitude of a signal when probing blended sources in nearby pixels.

**Comment by barentsen** [comment]
Thank you for the feedback @MNGuenther!! :+1: 

I wonder if we want to enable this and other use cases by allowing the user to pass a generic `transform` function which is applied to the light curve.

Example:
```python
tpf.interact(transform=lambda lc: lc.normalize())
```

But perhaps normalization is a sufficiently common use case that we may just want to add a more user-friendly `transform` option instead:

```python
tpf.interact(normalize=True)
```

Any thoughts/comments/votes? /cc @gully

**Comment by barentsen** [comment]
Ping @gully, how do you feel about adding a generic `transform` argument to `tpf.interact` to address this request?

**Comment by gully** [comment]
This transform function could be modeled after the `corrector_func` argument of the new `stitch` method-- it takes in a lightcurve and returns a lightcurve.  Should be doable.  

Some conceivable issues: the time and cadence samples are hardcoded, so you probably couldn't make your `transform` function drop cadences; folding on a period could even get dicey since you're re-arranging time and the cadence slider wouldn't map in monotonic order.  So maybe the trick would be to only accept/return the flux vector of the lightcurve, rather than the lightcurve itself.

**Comment by barentsen** [comment]
It would be fair to place a constraint on the `transform` function (or `lc_transform`?) in that it must not change the cadence or time values.

This could be documented, and, if deemed to be a common failure mode, yield a helpful error.

**Comment by gully** [comment]
OK, I experimented with a solution:

![20190924_interact_transform](https://user-images.githubusercontent.com/860227/65536168-bba45000-deb7-11e9-9200-9bb177a9b94e.gif)

Thoughts?

**Comment by barentsen** [comment]
This looks really great to me!

I think the most important application for this may be to support systematics removal, e.g. I'd hope this will work
```python
transform_func=lambda lc: lc.to_corrector("sff").correct()
```

We should also support correctors that need the TPF, e.g. we should make sure something like this eventually works:
```python
transform_func=lambda lc : PLDCorrector(tpf, aperture_mask=lc.aperture_mask).correct()
```

**Comment by gully** [comment]
Confirmed that `transform_func=lambda lc: lc.to_corrector("sff").correct()` works!  It's a bit slow at the moment, but the new sff corrector class will make it much faster. 

Supporting TPF input would be a bit trickier, since the function needs to be aware of not only the TPF but also the interactively selected aperture mask.  The function would have to take in a tpf and an aperture_mask and return a lightcurve.  So maybe you could have `lc_transform=` and `tpf_transform=` but with better names.  The `tpf` prefix hints that you are doing something to alter the appearance of the tpf images, which would not be accurate.

**Comment by barentsen** [comment]
@gully The user has access to the tpf object (by definition, because they call `tpf.interact`), and we can provide access to the aperture mask by storing it in e.g. `lc.meta['aperture_mask']`, so I think this shouldn't be quite possible with a single transform function!

**Comment by barentsen** [comment]
The solution as explained above has been merged into master as part of #600, and will be part of the upcoming v1.5.0 release. :+1: Closing this issue!
